pwa/src/components/applications/applicationForm.tsx

```diff
@@ -0,0 +1,233 @@
+import {ArrayInputComponent} from "../common/arrayInput";
+import FlashMessage from 'react-flash-message';
+
+export default function ApplicationForm({ id }) {
+  const [context, setContext] = React.useState(null);
+  const [alert, setAlert] = React.useState(null);
+  const [application, setApplication] = React.useState(null);
+  const [showSpinner, setShowSpinner] = React.useState(false);
+
+  React.useEffect(() => {
+    if (typeof window !== "undefined" && context === null) {
+      setContext({
+        adminUrl: window.GATSBY_ADMIN_URL,
+        frontendUrl: window.GATSBY_FRONTEND_URL,
+      });
+    } else {
+      if (isLoggedIn()) {
+        if (id !== "new") {
+          getApplication();
+        }
+      }
+    }
+  }, [context]);
+
+  const getApplication = () => {
+    setShowSpinner(true);
+    fetch(`${context.adminUrl}/applications/${id}`, {
+      credentials: "include",
+      headers: { "Content-Type": "application/json", 'Authorization': 'Bearer ' + sessionStorage.getItem('jwt') },
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        setShowSpinner(false);
+        setApplication(data);
+      })
+      .catch((error) => {
+        setShowSpinner(false);
+        console.error("Error:", error);
+        setAlert(null);
+        setAlert({ type: 'danger', message: error.message });
+      });
+  };
+
+  const saveApplication = (event) => {
+    event.preventDefault();
+    setShowSpinner(true);
+
+    let domains = retrieveFormArrayAsOArray(event.target, "domains");
+
+    let body = {
+      name: event.target.name.value,
+      description: event.target.description.value,
+      public: event.target.public.value,
+      secret: event.target.secret.value,
+      resource: event.target.resource.value,
+    };
+
+    if (domains.length !== 0) {
+      body["domains"] = domains;
+    } else {
+      body["domains"] = [];
+    }
+
+    body = removeEmptyObjectValues(body);
+    if (!checkValues([body.name])) {
+      return;
+    }
+
+    let url = `${context.adminUrl}/applications`;
+    let method = "POST";
+    if (id !== "new") {
+      url = `${url}/${id}`;
+      method = "PUT";
+    }
+
+    fetch(url, {
+      method: method,
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: "Bearer " + sessionStorage.getItem("jwt"),
+      },
+      body: JSON.stringify(body),
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        setShowSpinner(false);
+        setApplication(data);
+        method === 'POST' && navigate(`/applications`)
+      })
+      .catch((error) => {
+        setShowSpinner(false);
+        console.error("Error:", error);
+        setAlert(null);
+        setAlert({ type: 'danger', message: error.message });
+      });
+  };
+
+  return (<>
+      {
+        alert !== null &&
+        <FlashMessage duration={5000}>
+          <Alert alertClass={alert.type} body={function () { return (<>{alert.message}</>) }} />
```

Comment:
Hier is een custom wrapper gebouwd om een Bootstrap `Alert`, waardoor die onnodig complex wordt.

We kunnen `react-bootstrap` installeren (via npm) en dan op de volgende manier een `Alert` aanmaken: 

```
<Alert key={idx} variant='primary'>
  {alert.message}
</Alert>
```

_Ook out of scope voor deze PR, maar: deze Alert wordt nu ingeladen vanuit ons NL-Design-System, waar als beschrijving staat dat dit een bootstrap alert is, maar hier wordt geen bootstrap in gebruikt_